frb_rust/src/rust2dart.rs

```diff
@@ -111,3 +112,13 @@ impl<T: IntoDart> StreamSink<T> {
         self.rust2dart.close_stream()
     }
 }
+
+pub trait BoxIntoDart {
```

Comment:
nit: Maybe add a line of comment mentioning why we have such weird trait, probably simply link to our discussion about it in github